subprojects/docs/src/samples/buildCache/cacheable-bundle/groovy/build.gradle

```diff
@@ -15,13 +15,9 @@ task bundle {
     outputs.file(bundle)
         .withPropertyName("bundle")
 
-    doLast {
-        bundle.withOutputStream { stream ->
-            scripts.listFiles().sort().each {
-                stream.write(it.bytes)
-            }
-        }
-    }
+    main = "org.gradle.sample.Bundle"
+    classpath = project.sourceSets.main.runtimeClasspath
+    args projectDir.toPath().relativize(scripts.toPath()).toString(), projectDir.toPath().relativize(bundle.toPath()).toString()
```

Comment:
I think we should not promote using of relative paths for the JavaExec task. We have `CommandLineArgumentProvider` and `Exec.argumentProviders.add` exactly for that reason. I think we should leverage `CommandLineArgumentProvider` in this example.

Actually, we need to ignore the path to the executable as well (at least as long as something like https://github.com/gradle/gradle/pull/6711 has not been merged). If not, people are forced to install Java in the same directory. So why not declare a custom webpack task which uses `Project.javaExec`? The other option would be to subclass the `Exec` task and override the `getExecutable()` method annotating it with `@Internal`.

And the final option is to override the executable property via the runtime API by doing something like `inputs.property("executable", "webpack")`.

I would prefer using the custom task class and `Project.javaExec`.

Reply:
OK, I'll go with a Webpack task. 

For the purpose of this example, it will not be cacheable (of course), and will take two files as arguments, but these arguments won't be inputs, otherwise I will have the same problem as for the JavaExec task: the task will have inputs that I can't remove (AFAIK), and that will be absolute paths, and thus make the task not relocatable. 

Given that the task won't even appear in the guide, there is no need to use javaExec, right? I can just use groovy/kotlin code directly in the task as I did in my first attempt.

Reply:
Actually, to be more realistic, I think it would be more logical to use a very generic NpmTask, that would just take a list of arguments. A specific WebpackTask would probably define inputs and outputs, making the example look weird.